packages/firestore/src/core/firestore_client.ts

```diff
@@ -15,6 +15,12 @@
  */
 
 import { CredentialsProvider } from '../api/credentials';
+import {
+  DocumentReference,
+  DocumentSnapshot,
+  Query,
+  QuerySnapshot
+} from '../api/database';
```

Comment:
I think we want to avoid api/ dependencies here (CredentialsProvider is kind of an exception). The api/ layer should be responsible to converting to/from model types.

FWIW, I think as-written this may be impossible to port to Android, since you can't have circular dependencies in java.